src/snowflake/cli/plugins/nativeapp/artifacts.py

```diff
@@ -16,16 +16,31 @@
 
 import itertools
 import os
+from dataclasses import dataclass
 from pathlib import Path
 from textwrap import dedent
-from typing import Any, Callable, Dict, Iterable, Iterator, List, Optional, Tuple, Union
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Callable,
+    Dict,
+    Iterable,
+    Iterator,
+    List,
+    Optional,
+    Tuple,
+    Union,
+)
 
 from click.exceptions import ClickException
 from snowflake.cli.api.constants import DEFAULT_SIZE_LIMIT_MB
 from snowflake.cli.api.project.schemas.native_app.path_mapping import PathMapping
 from snowflake.cli.api.secure_path import SecurePath
 from yaml import safe_load
 
+if TYPE_CHECKING:
+    from snowflake.cli.plugins.nativeapp.project_model import NativeAppProjectModel
```

Comment:
Why do we need to use `if TYPE_CHECKING`?

Reply:
+1 that's surprising

Reply:
It's a way to handle cyclic type dependencies. I wanted to specify the input type of `BundleContext.from_na_project()`, but it created a module dependency (`NativeAppProjectModel` depends on this module).

Reply:
Why does NAPM depend on BundleContext? I don't see why the cycle needs to exist.

Reply:
It depends on the `artifacts.py` module, not the class. I will extract `BundleContext` to its own file to break the cycle.